x-pack/platform/plugins/private/intercepts/public/prompter/prompter.ts

```diff
@@ -70,27 +77,64 @@ export class InterceptPrompter {
       .pipe(Rx.filter((response) => !!response))
       .pipe(
         Rx.mergeMap((response) => {
-          const now = Date.now();
+          // Define safe timer bound at 24 days, javascript browser timers are not reliable for longer intervals
+          // see https://developer.mozilla.org/en-US/docs/Web/API/Window/setTimeout#maximum_delay_value, rxjs can do longer intervals, but we want to avoid
+          // the risk of running into issues with browser timers.
+          const safeTimerInterval = 24 * 24 * 60 * 60 * 1000; // 24 days in milliseconds
```

Comment:
I think worth extracting this part into `safeTime` operator and keep only business logic here. 

```
const MAX_TIMEOUT = 0x7fffffff;              // 2 147 483 647 ms
function safeDelay(ms: number): Observable<number> {
  return ms <= MAX_TIMEOUT
    ? timer(ms)                              // one shot
    : timer(MAX_TIMEOUT).pipe(
        ignoreElements(),                    // wait silently…
        concatWith(safeDelay(ms - MAX_TIMEOUT))
      );
}
```

Reply:
Not a suggestion for the reuse purpose, just find it weird that this.MAX_TIMER_INTERVAL concern is mixed together with business logic